### ISSUE TITLE
Fix FP & FN S2189 (`no-infinite-loop`): 

### ISSUE THREAD

**Issue by guillaume-dequenne-sonarsource** [SonarSource/SonarJS#2399]
Rule S2189 now relies on eslint [no-unmodified-loop-condition](https://eslint.org/docs/rules/no-unmodified-loop-condition) rather than from the symbolic execution of the Java-based frontend. The scope of this rule is also a bit different from ours, as it reports issues on any loop that uses a constant condition as part of its test, even if the test condition itself is not constant.

This is the source of false positives and false negatives that should be fixed. For example:

```
function do_while() {
  var trueValue = true;      
  do {                      // FN
    trueValue = true;
  } while (trueValue);
}
```

```
function someFunction(shouldExecute) {
  var i = 0;
  while (shouldExecute && i<5) { // FP: shouldExecute is not modified is the loop, but the loop is not infinite.
    console.log("hello");
    i++;
  }
}
```

**Comment by ilia-kebets-sonarsource** [comment]
this seems hard.

**Comment by yassin-kammoun-sonarsource** [comment]
Yes, it is hard. I think we can keep the ticket and check whether the JavaScript engine developed by R&D will be available to help us in that matter.

**Comment by ilia-kebets-sonarsource** [comment]
related to another issue for this rule https://github.com/SonarSource/SonarJS/issues/2410

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-132
